platform/ATCmdParser.h

```diff
@@ -222,7 +222,23 @@ class ATCmdParser : private NonCopyable<ATCmdParser> {
      */
     bool recv(const char *response, ...) MBED_SCANF_METHOD(1, 2);
 
-    bool vrecv(const char *response, std::va_list args);
+    /**
```

Comment:
I would say there is no necessary need to change the API, the original functions could act as wrappers for the newly introcuded `vrecvscanf`

Reply:
`vrecv` and `vscanf`:
* were already wrapped by `recv` and `scanf`, 
* were not used anywhere else, but in `recv` and `scanf`
* were not documented properly with Doxygen
I understand they were mentioned in the header file, but I don't think they were ever really meant to be an API?
I would go with Marcin's solution - they can be safely removed.
But actually - should we perhaps move the function header and the doxygen of `vrecvscanf` into the *.cpp file, to make this function private and pretend to be an API any more?
@mtomczykmobica , @VeijoPesonen , would that make sense?

Reply:
Agree with @michalpasztamobica. There is no reason to add another level of calls. recv and scanf are API and wrap old vrecv and vscanf/new vrecvscanf.

Reply:
Those are kind of points @kjbracey-arm should answer. My personal opinion is that once it's exposed as part of public API, undocumented or not, you can't remove it like that. Least what should be done is to mark the PR as breaking change. And yes, if the API is to be changed then the new function should be made as private.

Reply:
@VeijoPesonen agree vrecvscanf should be private. Changes done

Reply:
After consultation with @michalpasztamobica  functions vrecv and vscanf come back to code. Function vrecvscanfk made private. PR type doesn't change.